packages/astro/src/core/config/settings.ts

```diff
@@ -134,3 +145,64 @@ export async function createSettings(config: AstroConfig, cwd?: string): Promise
 
 	return settings;
 }
+
+function validateTsconfig(settings: AstroSettings, logger: Logger, rawConfig: TSConfig) {
+	try {
+		const { typescript } = settings.config.experimental;
+		if (!typescript) {
+			return;
+		}
+
+		let newConfig = { ...rawConfig };
+
+		if (!rawConfig.extends) {
+			newConfig.extends = ['astro/tsconfigs/base', GENERATED_TSCONFIG_PATH];
```

Comment:
It's okay to not use our presets, I'd suggest adding the preset only if there's no tsconfig at all